bitsandbytes/autograd/_functions.py

```diff
@@ -564,7 +568,7 @@ def matmul(
 
 def matmul_4bit(A: torch.Tensor, B: torch.Tensor, quant_state: F.QuantState, out: Optional[torch.Tensor] = None, bias=None):
     assert quant_state is not None
-    if A.numel() == A.shape[-1] and A.requires_grad == False:
+    if A.numel() == A.shape[-1] and A.requires_grad == False and A.device.type == "cuda":
```

Comment:
For other devices, they should also call this function but then redirect in the MatMul4Bit to the appropriate library call. This means we cannot have a check `A.device.type == "cuda"` at this point. Let me know if I am missing something here.

Reply:
When actual support for other devices/backends is added, we'd likely go through the codebase with a `"cuda"`-toothed comb (😁) and excise checks like this.

Reply:
This change was thinking of `gemv_4bit` (fusion optimizations of "Dequant + Matmul") right now is only limited to "cuda". Will remove this change first and whether it is needed may depend on the condition when adding a new device.